process/block/preprocess/transactions_test.go

```diff
@@ -540,6 +543,138 @@ func TestTransactionPreprocessor_RemoveBlockTxsFromPoolOK(t *testing.T) {
 	assert.Nil(t, err)
 }
 
+func TestTransactions_CreateAndProcessMiniBlockCrossShardGasLimitAddAll(t *testing.T) {
+	t.Parallel()
+
+	txPool, _ := shardedData.NewShardedData(storageUnit.CacheConfig{Size: 100000, Type: storageUnit.LRUCache})
+	requestTransaction := func(shardID uint32, txHashes [][]byte) {}
+	hasher := &mock.HasherMock{}
+	marshalizer := &mock.MarshalizerMock{}
+
+	txs, _ := NewTransactionPreprocessor(
+		txPool,
+		&mock.ChainStorerMock{},
+		hasher,
+		marshalizer,
+		&mock.TxProcessorMock{ProcessTransactionCalled: func(transaction *transaction.Transaction, round uint64) error {
+			return nil
+		}},
+		mock.NewMultiShardsCoordinatorMock(3),
+		&mock.AccountsStub{},
+		requestTransaction,
+	)
+	assert.NotNil(t, txs)
+
+	sndShardId := uint32(0)
+	dstShardId := uint32(1)
+	strCache := process.ShardCacherIdentifier(sndShardId, dstShardId)
+
+	addedTxs := make([]*transaction.Transaction, 0)
+	for i := 0; i < 10; i++ {
+		newTx := &transaction.Transaction{GasLimit: uint64(i)}
+
+		txHash, _ := core.CalculateHash(marshalizer, hasher, newTx)
+		txPool.AddData(txHash, newTx, strCache)
+
+		addedTxs = append(addedTxs, newTx)
+	}
+
+	mb, err := txs.CreateAndProcessMiniBlock(sndShardId, dstShardId, process.MaxItemsInBlock, haveTimeTrue, 10)
+	assert.Nil(t, err)
+
+	assert.Equal(t, len(addedTxs), len(mb.TxHashes))
+}
+
+func TestTransactions_CreateAndProcessMiniBlockCrossShardGasLimitAddAllAsNoSCCalls(t *testing.T) {
+	t.Parallel()
+
+	txPool, _ := shardedData.NewShardedData(storageUnit.CacheConfig{Size: 100000, Type: storageUnit.LRUCache})
+	requestTransaction := func(shardID uint32, txHashes [][]byte) {}
+	hasher := &mock.HasherMock{}
+	marshalizer := &mock.MarshalizerMock{}
+
+	txs, _ := NewTransactionPreprocessor(
+		txPool,
+		&mock.ChainStorerMock{},
+		hasher,
+		marshalizer,
+		&mock.TxProcessorMock{ProcessTransactionCalled: func(transaction *transaction.Transaction, round uint64) error {
+			return nil
+		}},
+		mock.NewMultiShardsCoordinatorMock(3),
+		&mock.AccountsStub{},
+		requestTransaction,
+	)
+	assert.NotNil(t, txs)
+
+	sndShardId := uint32(0)
+	dstShardId := uint32(1)
+	strCache := process.ShardCacherIdentifier(sndShardId, dstShardId)
+
+	gasLimit := process.MaxGasLimitPerMiniBlock / uint64(5)
+
+	addedTxs := make([]*transaction.Transaction, 0)
+	for i := 0; i < 10; i++ {
+		newTx := &transaction.Transaction{GasLimit: gasLimit, GasPrice: uint64(i), RcvAddr: []byte("012345678910")}
+
+		txHash, _ := core.CalculateHash(marshalizer, hasher, newTx)
+		txPool.AddData(txHash, newTx, strCache)
+
+		addedTxs = append(addedTxs, newTx)
+	}
+
+	mb, err := txs.CreateAndProcessMiniBlock(sndShardId, dstShardId, process.MaxItemsInBlock, haveTimeTrue, 10)
+	assert.Nil(t, err)
+
+	assert.Equal(t, len(addedTxs), len(mb.TxHashes))
```

Comment:
Why does this assert work? You add 10 txs, each with maxGasLimitPerMiniblock/5 => you add in the miniblock 2*maxGasLimitPerMiniblock. Is that ok?

Reply:
because these txs are not smart contract calls, normal transactions are processed fast, it is enough to add the constant.

Reply:
but you still have more gas than the maxGasLImitPerMiniBlock... it shouldn't allow the gasLimit to be more than the max

Reply:
ok